From 95bd054e827947d670b3c800fceab60be9666a34 Mon Sep 17 00:00:00 2001
From: janEbert <janpublicebert@posteo.net>
Date: Thu, 6 Feb 2020 23:32:12 +0100
Subject: [PATCH] Add Git stash amount to agnoster prompt

---
 themes/agnoster.zsh-theme | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/themes/agnoster.zsh-theme b/themes/agnoster.zsh-theme
index 63452378..c54084ce 100644
--- a/themes/agnoster.zsh-theme
+++ b/themes/agnoster.zsh-theme
@@ -177,6 +177,14 @@ prompt_git_relative() {
   fi;
 }
 
+# From https://lobste.rs/s/x5ioqm/what_does_your_shell_prompt_look_like#c_eejehg
+function git_stash_count() {
+    count=$(git stash list 2> /dev/null | wc -l) || return
+    if [[ $count -gt 0 ]]; then
+        echo -n " (${count})"
+    fi
+}
+
 # Git: branch/detached head, dirty status
 prompt_git() {
   (( $+commands[git] )) || return
@@ -236,6 +244,7 @@ prompt_git() {
     vcs_info
     echo -n "${${ref:gs/%/%%}/refs\/heads\//$PL_BRANCH_CHAR }${vcs_info_msg_0_%% }${mode}"
     [[ $AGNOSTER_GIT_INLINE == 'true' ]] && prompt_git_relative
+    git_stash_count
   fi
 }
 
-- 
2.34.1

