From 27c633691ecb83c56e1861c6f4cfc1f4ee52fc3a Mon Sep 17 00:00:00 2001
From: janEbert <janpublicebert@posteo.net>
Date: Thu, 6 Feb 2020 23:32:12 +0100
Subject: [PATCH] Add git stash amount to agnoster prompt

---
 themes/agnoster.zsh-theme | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/themes/agnoster.zsh-theme b/themes/agnoster.zsh-theme
index 8c700d06..eadbba93 100644
--- a/themes/agnoster.zsh-theme
+++ b/themes/agnoster.zsh-theme
@@ -93,6 +93,14 @@ prompt_context() {
   fi
 }
 
+# From https://lobste.rs/s/x5ioqm/what_does_your_shell_prompt_look_like#c_eejehg
+function git_stash_count() {
+    count=$(git stash list 2> /dev/null | wc -l) || return
+    if [[ $count -gt 0 ]]; then
+        echo -n " (${count})"
+    fi
+}
+
 # Git: branch/detached head, dirty status
 prompt_git() {
   (( $+commands[git] )) || return
@@ -136,6 +144,7 @@ prompt_git() {
     zstyle ':vcs_info:*' actionformats ' %u%c'
     vcs_info
     echo -n "${ref/refs\/heads\//$PL_BRANCH_CHAR }${vcs_info_msg_0_%% }${mode}"
+    git_stash_count
   fi
 }
 
-- 
2.20.1

